src/ILCompiler.Compiler/src/Compiler/ExportedMethodsRootProvider.cs

```diff
@@ -36,11 +46,44 @@ public void AddCompilationRoots(IRootingServiceProvider rootProvider)
                     if (ecmaMethod.IsNativeCallable)
                     {
                         string nativeCallableExportName = ecmaMethod.GetNativeCallableExportName();
+
                         if (nativeCallableExportName != null)
+                        {
+                            _methods.Add(ecmaMethod);
                             rootProvider.AddCompilationRoot(method, "Native callable", nativeCallableExportName);
+                        }
                     }
                 }
             }
+
+            // _exportsFile is not null when it's a shared library build
+            if (_exportsFile != null)
+                EmitExportedMethods();
+        }
+
+        private void EmitExportedMethods()
+        {
+            string moduleName = Path.GetFileNameWithoutExtension(_exportsFile);
+            FileStream fileStream = new FileStream(_exportsFile, FileMode.OpenOrCreate);
```

Comment:
This should be `FileMode.Create`. We want to overwrite the file if it exists already.